File: projects/JavaFXApps/BotManager/BotLib/src/main/java/net/rickiekarp/botlib/plugin/BotSetting.java
package net.rickiekarp.botlib.plugin;

import javafx.scene.Node;

public class BotSetting {
    private final String name;
    private final String desc;
    private boolean isVisible;
    private Node node;

    public BotSetting(Builder builder) {
        this.name = builder.name;
        this.desc = builder.desc;
        this.isVisible = builder.isShown;
        this.node = builder.node;
    }

    public String getTitle() {
        return name;
    }

    public String getDesc() {
        return desc;
    }

    public boolean isVisible() {
        return isVisible;
    }

    public void setVisible(boolean isVisible) {
        this.isVisible = isVisible;
    }

    public Node getNode() {
        return node;
    }

    public static final class Builder {
        private String 			name;
        private String 			desc;
        private boolean 			isShown;
        private Node node;

        public static Builder create() {
            return new Builder();
        }

        public Builder setName(final String s) {
            name = s;
            return this;
        }

        public Builder setDescription(final String s) {
            desc = s;
            return this;
        }

        public Builder setVisible(final boolean b) {
            isShown = b;
            return this;
        }
        public Builder setNode(final Node n) {
            node = n;
            return this;
        }


        public BotSetting build() {
            if (this.node == null) {
                throw new RuntimeException("NetworkAction requires a host");
            }
            return new BotSetting(this);
        }
    }

}

File: projects/JavaFXApps/BotManager/BotLib/src/main/java/net/rickiekarp/botlib/plugin/BotSetting.kt
package net.rickiekarp.botlib.plugin

import javafx.scene.Node

class BotSetting(builder: Builder) {
    val title: String?
    val desc: String?
    var isVisible: Boolean = false
    val node: Node?

    init {
        this.title = builder.nameString
        this.desc = builder.descString
        this.isVisible = builder.isShownBoolean
        this.node = builder.nodeBuilder
    }

    class Builder {
        var nameString: String? = null
        var descString: String? = null
        var isShownBoolean: Boolean = false
        var nodeBuilder: Node? = null

        fun setName(s: String): Builder {
            nameString = s
            return this
        }

        fun setDescription(s: String): Builder {
            descString = s
            return this
        }

        fun setVisible(b: Boolean): Builder {
            isShownBoolean = b
            return this
        }

        fun setNode(n: Node): Builder {
            nodeBuilder = n
            return this
        }


        fun build(): BotSetting {
            if (this.nodeBuilder == null) {
                throw RuntimeException("NetworkAction requires a host")
            }
            return BotSetting(this)
        }

        companion object {

            fun create(): Builder {
                return Builder()
            }
        }
    }

}
